### DISCUSSION TITLE
HVAC/Plumbing modeling tools

### DISCUSSION THREAD

**Discussion by Hauk-Lykke** [IfcOpenShell/IfcOpenShell#4172] [Ideas]
I have five tools i would love to get into BlenderBIM that Revit(with MagiCAD) currently has, that in my opinion are necessary for the MEP-modeling to be able to compete with Revit.

1. Add a pipe/duct branch by simply clicking one click where each corner/end point will be. This is basically drawing a line with many segments that converts to pipes/ducts with fittings. Making four correctly connected pipes thus would take 5 mouse clicks.

2. Standard connections. Two pipes or ducts that need two fittings to connect because they are lined up in different planes should be able to connect with standard fittings (typically 45 or 90 degree elbows). In MagiCAD the user here normally gets 1-9 options for how this can best be solved, and then the user picks the best option.

3. Parallell pipes with insulation or a "pipe segment group" concept, where the user can make a group of pipe segment types and model with all of them simultaneously. In big buildings, pipes often run alongside 4-6 other pipes in the main routings. Instead of drawing 8 almost identical pipe layouts with 8 different pipe types like the user has to in Revit, it should be possible to make one and then just draw all the others offset from this one and then change the types afterward. It's almost possible in Revit already, the paralell pipes feature exists. But as long as the insulation types and pipe types are incredibly cumbersome to change, it's often easier to just draw each system separately. There has to be a better way!

4. Pipe insulation standards. This is the ability to set a table where the user can select a fluid temperature and insulation type, and from that automatically get the correct insulation thickness for each pipe segment.

5. Pressure drop calculations and pressure drop balancing with calculation reports.

**Comment by hydromac** [comment]
I am an engineer designing water, sewage and hvac installations.  I know the basics of object-oriented design in python and I am interested in software development in the direction you mentioned.  Can I ask you to contact with me hydromac'at'wp.pl?

**Reply by Moult** [reply]
Sent email :)

**Comment by Isaina** [comment]
There is a more info

https://github.com/IfcOpenShell/IfcOpenShell/issues/2181

**Comment by Moult** [comment]
We're currently in the process of implementing the rules that generate fittings (i.e. step 2). Would you like to help build / test / report bugs / sponsor development?

**Reply by hydromac** [reply]
@Moult Sorry I didn't reply for a long time (I had health problems). I am interested in helping build/test/bug report HVAC & plumbing development.
